src/plugins/intel_gpu/src/graph/reorder.cpp

```diff
@@ -188,7 +188,18 @@ std::vector<layout> reorder_inst::calc_output_layouts(reorder_node const& /*node
             onednn_weights_params->_in_desc = onednn::layout_to_memory_desc(input_layout);
         }
 #endif // ENABLE_ONEDNN_FOR_GPU
-        return { desc->weights_reorder_params->get_output_layout() };
+        if (node.get_users().size() != 0 && node.have_user_with_type<convolution>()) {
+            auto input_laytout = node.get_users().front()->get_dependency(0).get_output_layout();
+            auto weights_layout = desc->weights_reorder_params->get_output_layout();
+            return { input_layout.get_partial_shape().size() == 3
+                     ? layout(ov::PartialShape{weights_layout.batch(), weights_layout.feature(), weights_layout.spatial(1) * weights_layout.spatial(0)},
+                              weights_layout.data_type,
+                              weights_layout.format,
+                              weights_layout.data_padding)
+                     : desc->weights_reorder_params->get_output_layout() };
```

Comment:
Please fix weights_reorder_params->get_output_layout() to return expected layout instead of fixing reorder

Reply:
Not fixed as I mentioned. I meant that to fix where the weights_Reorder_params is set. (i.e., in convolution kernel)

Reply:
Tried to update as you commented. Thanks for your comment. Thanks @sshlyapn

Reply:
@steve-y, can partial shape be obtained directly from node's second input_layout?